x-pack/platform/plugins/shared/onechat/server/services/tools/tool_types/esql/create_handler.ts

```diff
@@ -17,8 +17,28 @@ export const createHandler = (
 ): ToolHandlerFn<z.infer<ZodObject<any>>> => {
   return async (params, { esClient }) => {
     const client = esClient.asCurrentUser;
+
+    // Apply default values for parameters that weren't provided by the LLM
+    const resolvedParams = Object.keys(configuration.params).reduce((acc, paramName) => {
+      const param = configuration.params[paramName];
+      const providedValue = params[paramName];
+
+      if (providedValue !== undefined) {
+        // LLM provided a value, use it
+        acc[paramName] = providedValue;
+      } else if (param.optional && param.defaultValue !== undefined) {
+        // LLM didn't provide a value, but we have a default
+        acc[paramName] = param.defaultValue;
+      } else {
+        // No value provided and no default, use null
+        acc[paramName] = null;
+      }
+
+      return acc;
+    }, {} as Record<string, unknown>);
+
```

Comment:
NIT: you could extract that to its own function, `(configParams, inputParams) => resolvedParams`

Would also make testing easier, you wouldn't need to test through the handler as you did. (even if you also did add tests for other aspects of the handler - and thanks for that)

Reply:
https://github.com/elastic/kibana/pull/238008/commits/1c97fa27c6a90875a2f30e3f3d13819fcebe9060